packages/nightingale-interpro-track/src/nightingale-interpro-track.ts

```diff
@@ -258,6 +302,51 @@ class NightingaleInterproTrack extends NightingaleTrack {
     return expanded ? this.getFeatureColor(f) : "white";
   }
 
+  private getTextLabel(datum: Feature): string | null {
+    if (this.label?.length) {
+      if (this.label.startsWith(".")) {
+        return _get(datum, this.label.slice(1), null);
+      }
+      return this.label;
+    }
+    return datum?.feature?.accession || null;
+  }
+
+  private refreshLabels(base: LabelGroup) {
+    base
+      .attr("x", (f) => {
+        const start = f.start || 0;
+        const end = f.end || 0;
+        return this.getXFromSeqPosition(start + (end - start) / 2 || 1);
+      })
```

Comment:
This causes the label to shift out of the viewport when zooming.

Something like this could help keeping the label visible:

```js
const displayStart = this["display-start"] || 0;
const displayEnd = this["display-end"] || this.length;
const start = Math.max(f.start || 0, (displayStart as number));
const end = Math.min(f.end || 0, (displayEnd as number));
return this.getXFromSeqPosition(start + (end - start) / 2 || 1);
```

The `wrap()` function would have to be modified.